src/org/rascalmpl/values/RascalValueFactory.java

```diff
@@ -850,14 +858,19 @@ protected ITree wrap(ITree content, io.usethesource.capsule.Map.Immutable<String
         }
 	}
 	
-	private static class Cycle implements ITree, IExternalValue {
+	private static class Cycle implements ITree {
 		protected final IConstructor symbol;
 		protected final int cycleLength;
 		
 		public Cycle(IConstructor symbol, int cycleLength) {
 			this.symbol = symbol;
 			this.cycleLength = cycleLength;
 		}
+
+		@Override
+		public int getConcreteMatchFingerprint() {
+			return 95131878 /* "cycle".hashCode() */ + 13 * symbol.hashCode();
```

Comment:
should we not use the fingerprint of the `symbol`?

Reply:
No definitely not.

Reply:
okay 👍🏼